knode/knode.go

```diff
@@ -0,0 +1,77 @@
+package knode
+
+import (
+	"errors"
+	"os"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/client-go/kubernetes"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/record"
+
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+)
+
+var Kubeconfig *string
+
+type K8sNotifier struct {
+	nodeName  string
+	config    *rest.Config
+	recorder  record.EventRecorder
+	inCluster bool
+}
+
+func NewK8sNotifier(deploymentMode bool) *K8sNotifier {
+	return &K8sNotifier{
+		inCluster: deploymentMode,
+	}
+}
+
+// InitializeK8sNotifier sets up the event recorder to send events to the Kube API
+// server.
+func (k *K8sNotifier) InitializeK8sNotifier() error {
+	var err error
+	if k.inCluster {
+		k.config, err = rest.InClusterConfig()
+		if err != nil {
+			return err
+		}
+		k.nodeName = os.Getenv("NODE_NAME")
+		if err != nil {
```

Comment:
I think it should be k.nodeName == ""

Reply:
You are right.. Thanks for pointing out. 😃